src/gtc/oir.py

```diff
@@ -132,31 +131,35 @@ class Temporary(FieldDecl):
     pass
 
 
+def _check_interval(instance: Union["Interval", "UnboundedInterval"]) -> None:
```

Comment:
This function is only used inside a root_validators, right? So you could  also do something like:
```python
@datamodels.root_validator
@classmethod
def _check_interval(cls: Type["Interval"], instance: Union["Interval", "UnboundInterval"]) -> None:
    ....
```
and then assign it to some name inside the two classes where it is needed

Reply:
Right. I tried this and got
```
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

src/gtc/oir.py:138:72: F821 undefined name 'UnboundInterval'

mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

src/gtc/oir.py:136:2: error: "classmethod" used with a non-method  [misc]
src/gtc/oir.py:138:72: error: Name "UnboundInterval" is not defined  [name-defined]
src/gtc/oir.py:192:9: error: Missing positional argument "instance" in call to "_check_interval"  [call-arg]
src/gtc/oir.py:192:25: error: Argument 1 to "_check_interval" has incompatible type "UnboundedInterval"; expected "Type[Interval]"  [arg-type]
```